code/drasil-code/lib/Language/Drasil/Code/Imperative/DrasilState.hs

```diff
@@ -260,14 +261,14 @@ getExpInputFormat n chs _ = fMod (modularity chs) (inputStructure chs)
         fMod _ Bundled = []
         fMod Unmodular _ = [(giNm, n)]
         fMod (Modular Combined) _ = [(giNm, "InputParameters")]
-        giNm = "get_input"
+        giNm = showChsStr GetInput
```

Comment:
Here the question becomes: could the list of pairs being returned not have something more meaningful in its first argument? If it was changed, how many places would be affected?

Reply:
You are referring to the `n` in `getExpInputFormat n chs _`, correct?

Reply:
I'm referring to `giNm` still being a `String`. Could it be `GetInput`, for example, instead? Why make it into a `String` so eagerly?

Reply:
`getExpInputFormat` is expecting to return `[ModExp]`, a (String, String). 

Setting `giNm = GetInput` would result in the following error:

```
Couldn't match type ‘ComponentName’ with ‘[Char]’
Expected type: [ModExp]
Actual type: [(ComponentName, [Char])]
```

This would work if `GetInput` were a `String` or `Name`. We would not need to print using `showChsStr GetInput`. Is this the direction to go?

Reply:
The same applies here as below: maybe the type should be `RenderChoicesStr a => [(a, String)]` (where, of course, the second `String` will likely have its existence as a string questioned in the future.)

Another possibility is to have the type be `[(ComponentName, String)]`. In the short-term, that might be simpler.

Reply:
I'm not familiar with what is suggested here. Should this be using an `Either` instead somewhere? The following has more errors than the `RenderChoicesStr a => a` that was suggested below, so I must be misunderstanding how this is to be applied, or maybe it is not code but the context of what should happen.

```
getInOutCall :: (OOProg r, RenderChoicesStr a) => [(a, String)] -> GenState [CodeVarChunk] -> 
  GenState [CodeVarChunk] -> GenState (Maybe (MSStatement r))
```